server/src/test/java/io/crate/integrationtests/DiskThresholdDeciderIT.java

```diff
@@ -158,12 +158,12 @@ public void testHighWatermarkNotExceeded() throws InterruptedException, Executio
         // (subtract the translog size since the disk threshold decider ignores this and may therefore move the shard back again)
         fileSystemProvider.getTestFileStore(dataNode0Path).setTotalSpace(minShardSize + WATERMARK_BYTES - 1L);
         refreshDiskUsage();
-        assertThat(getShardRoutings(dataNode0Id, indexName)).isEmpty();
+        assertBusy(() -> assertThat(getShardRoutings(dataNode0Id, indexName)).isEmpty());
 
         // increase disk size of node 0 to allow just enough room for one shard, and check that it's rebalanced back
         fileSystemProvider.getTestFileStore(dataNode0Path).setTotalSpace(minShardSize + WATERMARK_BYTES + 1L);
         refreshDiskUsage();
-        assertThat(getShardRoutings(dataNode0Id, indexName)).hasSize(1);
+        assertBusy(() -> assertThat(getShardRoutings(dataNode0Id, indexName)).hasSize(1));
     }
```

Comment:
there is a [new test case](https://github.com/elastic/elasticsearch/commit/87076c32e219bd501be4a0dc1968fe7a4fe7b537#diff-e44f19e24fcf88d32bd88d5ed8c4f19b7eb5c01560fc20168784476143cef808R172) `testRestoreSnapshotAllocationDoesNotExceedWatermark` - I guess we need it as it's introduced in the linked patch and seems to be related?

Reply:
Added the test case